### DISCUSSION TITLE
Multiply roles and permission.

### DISCUSSION THREAD

**Discussion by VictorPulzz** [spatie/laravel-permission#1718] [Q&A] [answered]
Hi. I have a small question, is it possible to use your library to manage global and individual roles. For example:
let's say there is a tableusers and the "Add Tasks" global role, which determines whether the user can create tasks. But there is also a user role that depends on the board id, "Add tasks". Those. a user for a board cannot create a task. Something like this structure
`users` `roles` `permissions` `user_role` `user_board_permission`. I mean, the user has the global role "manager" (with the ability to create tasks), but for some board, he has the role "user" (without the ability to create tasks). This feature (https://spatie.be/docs/laravel-permission/v4/basic-usage/wildcard-permissions) can help, do you think ?
Thanks!

**Comment by drbyte** [comment, marked as answer]
As with any application, each layer needs to check and enforce its own rules for roles/permissions.

While the most common use of this package is for situations less complex than what you describe, there are many options for adding the extra layers you need.

Layer 1: Roles/Permissions per-user, enforced using `@can()` and `$someUser->can()`, middleware, etc. This may or may not include the wildcard-permissions feature, depending on needs.
Layer 2: optionally (only if relevant, and maybe not in your case) additional User-type models associated with additional Roles/Permissions
Layer 3: custom in-app relations between users and other models of your app, enforced using Model Policies, and/or middleware, and/or custom Gate rules. This may require constructing your own models, tables, foreign-key relationships, etc.

**Reply by VictorPulzz** [reply]
I have already found a way out of the situation, thanks for the answer :)
